Documentation/design/resource_dep.dot

```diff
@@ -0,0 +1,121 @@
+digraph resource {
+  size="8,6"; ratio=fill; node[fontsize=20];
+  compound=true;
+  node [shape=rectangle];
+
+  {
+    node [shape=plaintext];
+    init -> render -> prepare -> launch;
+  }
+
+  // User inputs.
+  subgraph cluster_input {
+    node [shape=rectangle,style=filled,color=gainsboro];
+    base_domain [label="base domain"];
+    cluster_name [label="cluster name"];
+    license [label="license"];
+    pull_secret [label="pull secret"];
+    platform [label="platform"];
+    email_address [label="email address"];
+    password [label="password"];
+  }
+
+  subgraph cluster_hint {
+    style=invis;
+    node [shape=rectangle,style=filled,color=lightblue];
+    output [label="output"];
+    node [shape=rectangle,style=filled,color=gainsboro];
+    input [label="input "];
+  }
+
+  // Install config.
+  {
+    node [style=filled,color=sandybrown];
+    install_config [label="install-config.yaml"];
+  }
+
+  // TLS assets
+  {
+    node [style=filled,color=violet];
+    tls_assets [label="tls assets"];
```

Comment:
Can you break this up into all of the TLS assets?

Reply:
I tried previously but there are 30 files, and even I bundle them in to crt key pair, there are still around 15, The result is pretty messy. I can give another try to reorganize them. BTW what's the best way to group them? Is there a way to zoom in zoom out ?

Reply:
Don't worry about it being messy. That's the job of the renderer. You could put them all into a subgraph cluster so that they are all arranged together, but I'm not sure that it's worth the effort. It might be easiest to just color nodes based on their datatype.

Reply:
[The rank approach][1] may break down with 15 TLS nodes (or we'll go really wide ;).  How about using `neato` to oraganize with node shape for type (tls, op config, ...) and fill color for phase?  Or gradations of color to separate types within a phase?

[1]: https://github.com/yifan-gu/installer/blob/6864274adee88c8b81a56378de1bb0f51dc4bf05/Documentation/design/resource_dep.dot#L116

Reply:
We could also color edges by command, since the installer commands are what's doing the conversion.